pkg/engine/engine.go

```diff
@@ -140,6 +143,16 @@ func (e *Engine) alterFuncMap(t *template.Template) template.FuncMap {
 		return buf.String()
 	}
 
+	// Add the 'required' function here
+	funcMap["required"] = func(warn string, val string) (string, error) {
```

Comment:
Yeah, it probably should.  I wrestled with this for a bit, and decided it'd be best to get feedback because I wasn't entirely sure.  I'll get this changed right quick :)